### PR TITLE
Parser bug: components no longer orphaned

### PR DESCRIPTION
**Pull request by kerimoyle**
Addresses #729

### CODE REVIEW

File: src/parser.cpp [hunk 1]
```diff
@@ -1074,9 +1074,15 @@ ComponentPtr Parser::ParserImpl::loadComponentRef(const ModelPtr &model, const X
             mParser->addIssue(issue);
         }
 
-        if ((parentComponent) && (childComponent)) {
+        if (childComponent) {
             // Set parent/child encapsulation relationship.
-            parentComponent->addComponent(childComponent);
+            if (parentComponent) {
```
Comment (agarny) [inline src/parser.cpp:1079]: Should read: `if (parentComponent != nullptr) {`.

**Review by agarny — CHANGES_REQUESTED** [review]
Just one minor change.

File: src/parser.cpp (hunk 1 above)
Comment (kerimoyle) [reply src/parser.cpp:1079]: OK ... then I'll also change the line above which reads `if(childComponent) {` too ... !

Comment (kerimoyle) [reply src/parser.cpp:1079]: Actually ... this is the idiom throughout the whole of the parser :(

Comment (agarny) [reply src/parser.cpp:1079]: Ok, forget about it then. Will have to be done as part of our big cleaning up.

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Pull request merged**